pythran/transformations/extract_doc_strings.py

```diff
@@ -0,0 +1,43 @@
+""" ExtractDocStrings fills a dictionnary with doc strings for each function. """
+
+from pythran.passmanager import Transformation
+
+import ast
+
+
+class ExtractDocStrings(Transformation):
+    '''
+    Extract Python Doc Strings, removing them from the AST and putting them in
+    a dictionary for later use.
+
+    >>> import ast
+    >>> from pythran import passmanager, backend
+    >>> node = ast.parse("def foo(): 'my doc is cool' ; pass")
+    >>> pm = passmanager.PassManager("test")
+    >>> _ = pm.apply(ExtractDocStrings, node)
+    >>> print pm.dump(backend.Python, node)
+    def foo():
+        pass
+    '''
+
+    def __init__(self):
+        super(ExtractDocStrings, self).__init__()
+        self.docstrings = dict()
+
+    def run(self, node, ctx):
+        super(ExtractDocStrings, self).run(node, ctx)
+        return self.docstrings
```

Comment:
> WHy not using the self.result?

it's not an analysis (it prunes some nodes from tha AST) we did the same
for NormalizeIdentifiers